Hospital/HospitalClassLib/SharedModel/Repository/Implementation/DoctorReadRepository.cs

```diff
@@ -21,22 +21,49 @@ public IEnumerable<Doctor> GetNonOverloadedDoctors()
         {
             var medRecords = _context.Set<MedicalRecord>().AsEnumerable();
 
+            var doctors = _context.Set<Doctor>()
+               .Where(d => d.Specialization.Name.ToLower().Equals("general practice")).AsEnumerable();
+
             var query = medRecords
                 .GroupBy(t => new { Doctor = t.DoctorId })
                 .Select(g => new { Count = g.Count(p => p.Id > 0), g.Key.Doctor });
 
-            var min = medRecords
-                .GroupBy(t => t.DoctorId)
-                .Select(g => g.Count()).Min();
+            var mrs = _context.Set<MedicalRecord>().Include(mr => mr.Doctor);
+            var docs = _context.Set<Doctor>();
+
+            var docsWithMR = (from mr in mrs where mr.Doctor != null select mr.Doctor);
```

Comment:
In one function we have a mix of query language and linq. Generally we would opt for one or the other. Preferably linq since it is code while this is just a feature in dotnet and requires somebody to learn the syntax. It is possible to change this with a statement mrs.Where(x => x.Doctor != null).Select(x=> x.Doctor).